action/protocol/execution/evm/evm.go

```diff
@@ -422,17 +422,18 @@ func executeInEVM(ctx context.Context, evmParams *Params, stateDB *StateDBAdapte
 		// After EIP-3529: refunds are capped to gasUsed / 5
 		refund = (evmParams.gas - remainingGas) / params.RefundQuotientEIP3529
 	}
-	// adjust refund due to dynamicGas
+	// before London EVM activation (at Okhotsk height), in certain cases dynamicGas
+	// has caused gas refund to change, which needs to be manually adjusted after
+	// the tx is reverted. After Okhotsk height, it is fixed inside RevertToSnapshot()
 	var (
-		refundLastSnapshot = stateDB.RefundAtLastSnapshot()
-		currentRefund      = stateDB.GetRefund()
-		featureCtx         = protocol.MustGetFeatureCtx(ctx)
+		deltaRefundByDynamicGas = evm.DeltaRefundByDynamicGas
```

Comment:
txcontext is not an ideal place to store the deltarefund. StateDB should be the only place to interact with evm from outside. we may add an interface to set delta refund value. Consequently, we can all stateDB.GetRefundWithDelta, and do not to fetch `feature context` here.

Reply:
the interface (SetDelta, GetRefundWithDelta) needs to be implemented in go-ethereum and iotex-core, but not necessary for go-ethereum. the current deltaRefund is just a temporary variable, I think there is no problem.

Reply:
TxContext.Origin and TxContext.GasPrice are also used in our iotex-core code, so i think it's fine to use it